packages/database/src/cachers/EntityHierarchyCacher.js

```diff
@@ -5,17 +5,126 @@
 import { reduceToDictionary } from '@tupaia/utils';
 import { ORG_UNIT_ENTITY_TYPES } from '../modelClasses/Entity';
 
+const REBUILD_DEBOUNCE_TIME = 1000; // wait 1 second after changes before rebuilding, to avoid double-up
+
 export class EntityHierarchyCacher {
   constructor(models) {
     this.models = models;
-    this.generationsVisited = new Set();
+    this.changeHandlerCancellers = [];
```

Comment:
Key changes in here:
- On any change of an entity or entity relation, we wipe the relevant subtrees, and kick off a rebuild
- We debounce and batch these rebuilds, so we aren't rebuilding on every individual change

Reply:
Good strategy